crates/sc-consensus-subspace-rpc/src/lib.rs

```diff
@@ -184,21 +184,34 @@ where
         + 'static,
     Client::Api: SubspaceRuntimeApi<Block, FarmerPublicKey>,
 {
-    fn get_farmer_metadata(&self) -> RpcResult<FarmerMetadata> {
+    fn get_farmer_protocol_info(&self) -> RpcResult<FarmerProtocolInfo> {
         let best_block_id = BlockId::Hash(self.client.info().best_hash);
         let runtime_api = self.client.runtime_api();
 
-        let farmer_metadata: Result<FarmerMetadata, ApiError> = try {
-            FarmerMetadata {
+        let genesis_hash = self
+            .client
+            .info()
+            .genesis_hash
+            .as_ref()
+            .try_into()
+            .map_err(|error| {
+                error!("Failed to convert genesis hash: {error}");
+                JsonRpseeError::Custom("Internal error".to_string())
```

Comment:
Why not display more info in the custom errror something like `JsonRpseeError::Custom(format!("Failed to convert genesis hash: {error}"))`?

Reply:
Because it is an RPC endpoint and error isn't related to the request. It is still printed in node's logs because it is a node's error.

Reply:
Well, it might not be a bad idea to show the error in the form of `Custom("Internal error: failed to convert genesis hash: {error}")`, I had an experience where calling an RPC directly returns an error like this without more context and I have to look into the code again. Just a thought, no strong opinion :P